src/components/molecules/PostList/index.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { PageSearchParams } from "@/types/pageSearchParams";
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { PostData } from "@/types/postData";
+import { useCallback, useEffect, useMemo, useRef } from "react";
+import { getPosts } from "@/apis/posts";
+import PostCard from "../PostCard";
+
+function PostList({ searchParams }: PageSearchParams) {
+  const getPostList = async ({ pageParam = 0 }, URLSearchParams: URLSearchParams) => {
+    const cityId = parseInt(URLSearchParams.get("cityId") || "0", 10);
+    const countryId = parseInt(URLSearchParams.get("countryId") || "0", 10);
+    const districtId = parseInt(URLSearchParams.get("districtId") || "0", 10);
+    const all = URLSearchParams.get("all");
+    const queryString = `?${cityId > 0 ? `cityId=${cityId}` : ""}${countryId > 0 ? `&countryId=${countryId}` : ""}${
+      districtId > 0 ? `&districtId=${districtId}` : ""
+    }${all === "true" || all === "false" ? `&all=${all}` : ""}`;
+    return getPosts(`${queryString}${pageParam ? `&key=${pageParam}` : ""}`);
+  };
+
+  const { data, fetchNextPage, hasNextPage } = useInfiniteQuery(
+    ["post_list", searchParams],
+    (reactQueryParam) => {
+      return getPostList(reactQueryParam, new URLSearchParams(searchParams));
+    },
+    {
+      retry: 2,
+      getNextPageParam: (lastPage) => {
+        const newKey = lastPage?.data?.response?.nextCursorRequest?.key;
+        return newKey !== -1 ? newKey : undefined; // 이전 페이지에서 받은 key값이 -1이면 undefined를 리턴하여 hasnextPage를 false로 설정
+      },
+    },
+  );
+
+  const target = useRef<HTMLDivElement>(null);
+
+  const handleIntersect = useCallback(
+    async ([entry]: IntersectionObserverEntry[], observer: IntersectionObserver) => {
+      if (entry.isIntersecting) {
+        observer.unobserve(entry.target);
+        if (hasNextPage) {
+          await fetchNextPage();
+        }
+        observer.observe(entry.target);
+      }
+    },
+    [fetchNextPage, hasNextPage],
+  );
+
+  const observer = useMemo(() => new IntersectionObserver(handleIntersect), [handleIntersect]);
+
+  useEffect(() => {
+    if (target.current) {
+      observer.observe(target.current);
+    }
+    return () => observer && observer.disconnect();
```

Comment:
observer는 항상 존재하는 변수인 것 같아요, && 연산은 제거하면 좋을 것 같습니다
```suggestion
    return () => observer.disconnect();
```

Reply:
수정했습니다